### PR TITLE
Fuzz test the trie building algorithm

### PR DESCRIPTION
**Pull request by zander-xyz**

### CODE REVIEW

File: include/monad/logging/formatter.hpp [hunk 1]
```diff
@@ -345,6 +346,62 @@ namespace fmt
         }
     };
 
+    template <>
+    struct formatter<monad::trie::Branch> : public monad::log::basic_formatter
+    {
+        template <typename FormatContext>
+        auto format(monad::trie::Branch const &branch, FormatContext &ctx) const
+        {
+            // TODO: ugly. We can avoid the byte wrapper together if we
```
Comment (zander-xyz) [inline include/monad/logging/formatter.hpp]: I plan on addressing this in a separate PR

Comment (zander-xyz) [reply include/monad/logging/formatter.hpp]: https://github.com/monad-crypto/monad/pull/134

File: cmake/test.cmake [hunk 2]
```diff
@@ -7,18 +7,26 @@ function(add_unit_test)
...
+    cmake_parse_arguments(
+        ADD_UNIT_TEST "${OPTIONAL_ARGS}" "${ONE_VALUE_ARGS}"
+        "${MULTI_VALUE_ARGS}" ${ARGN}
+    )
 
-    add_executable(${ADD_UNIT_TEST_TARGET} ${ADD_UNIT_TEST_SOURCES})
+    add_executable(
+        ${ADD_UNIT_TEST_TARGET}
+        ${ADD_UNIT_TEST_SOURCES}
+        ${PROJECT_SOURCE_DIR}/test/unit/common/src/test/main.cpp
```
Comment (rgarc) [inline cmake/test.cmake:18]: I guess I don't understand cmake, but how come adding this line didn't break the previous usages of `add_unit_test`?

Comment (zander-xyz) [reply cmake/test.cmake:18]: main was linked in via the common unit testing library

**Pull request merged**